api_client/python/timesketch_api_client/client.py

```diff
@@ -503,6 +504,67 @@ def upload(self, timeline_name, file_path):
             searchindex=timeline['searchindex']['index_name'])
         return timeline_obj
 
+    def upload_data_frame(
+            self, data_frame, timeline_name, format_message_string=''):
```

Comment:
Can we have format strings loaded from a YAML file? Maybe key off of the type (name) of the data?
Fine to put in a TODO and follow up in a later PR.